pipe.go

```diff
@@ -23,18 +25,23 @@ func (p *PipeReader) Close() error {
 	if p.core != nil {
 		loggerCore.DeleteCore(p.core)
 	}
-	return p.r.Close()
+	return multierr.Append(p.core.Sync(), p.closer.Close())
 }
 
 // NewPipeReader creates a new in-memory reader that reads from all loggers
 // The caller must call Close on the returned reader when done.
+//
+// By default, it:
+//
+// 1. Logs JSON.
+// 2. Logs at the Debug level. However, unless SetLogLevel is called on a
```

Comment:
I think it's better to set it to be the same as the primary logger's level by default. The user will then see the same logs that are being logged to stderr or the output file. They still have an option to inspect debug messages from subsystems that have had their log level decreased.

Either way the comment about "only error messages will be logged" is slightly confusing because if they have specified GOLOG_LOG_LEVEL=Info then the primary core will be emitting log events at that level and above.

Reply:
> I think it's better to set it to be the same as the primary logger's level by default.

It is. The primary log level is debug and _must_ be debug. Otherwise, calling `SetLogLevel("some-subsystem", "debug")` won't work because, while the messages will make it through the subsystem's logger, they'll get stopped by the primary logger because the primary logger will still filter out all messages with levels below ERROR.

(but I agree that the comment is really confusing)